src/sim_model.jl

```diff
@@ -285,6 +285,62 @@ function evaloutput(model::SimModel{NT}, d=model.buffer.empty) where NT <: Real
     return y
 end
 
+"""
+    savetime!(model::SimModel) -> t
+
+Set `model.t` to `time()`  and return the value.
+
+Used in conjunction with [`periodsleep`](@ref) for simple soft real-time simulations. Call
+this function before any other in the simulation loop.
+"""
+function savetime!(model::SimModel)
+    model.t[] = time()
+    return model.t[]
+end
+
+"""
+    periodsleep(model::SimModel, busywait=false) -> nothing
+
+Sleep for `model.Ts` s minus the time elapsed since the last call to [`savetime!`](@ref).
+
+It calls [`sleep`](https://docs.julialang.org/en/v1/base/parallel/#Base.sleep) if `busywait`
+is `false`. Else, a simple `while` loop implements busy-waiting. As a rule-of-thumb,
+busy-waiting should be used if `model.Ts < 0.1` s, since the accuracy of `sleep` is around 1
+ms. Can be used to implement simple soft real-time simulations, see the example below.
+
+# Examples
+```jldoctest
+julia> model = LinModel(tf(2, [0.3, 1]), 0.1);
+
+julia> function sim_realtime!(model)
+           t_0 = time()
+           for i=1:3
+               t = savetime!(model)      # first function called
+               println(round(t - t_0, digits=3))
+               updatestate!(model, [1])
+               periodsleep(model, true)  # last function called
+           end
+       end;
+
+julia> sim_realtime!(model)
+0.0
+0.1
+0.2
+```
+"""
+function periodsleep(model::SimModel, busywait=false)
+    if !busywait
+        model.Ts < 0.1 && @warn "busy-waiting is recommended for Ts < 0.1 s"
+        computing_time = time() - model.t[]
+        sleep_time = model.Ts - computing_time
+        sleep_time > 0 && sleep(sleep_time)
+    else
+        while time() - model.t[] < model.Ts
```

Comment:
This loop does no work and has no visible side effects, are you sure the compiler will not remove it, in some circumstances or at some point in the future?

Reply:
Huum I did not though of that but I would be really surprised.  There is no work in the body, but there is still the condition (`time() = model.t[] < model.Ts`) that need to be executed to figure out if the body needs to be executed, no matter what's inside the body.

Reply:
Also, I thought of adding the line `GC.enable(false)` in `savetime!`, and `GC.enable(true)` in `periodsleep`. It could be an additional argument e.g. `disable_gc` and `reactivate_gc`. The point is disabling garbage collecting except the controller is sleeping. I pretty sure that `sleep` allows the GC to run, but I'm not sure about the busy-waiting. Do you think it's a good idea ?

Otherwise, it could be just a warning in the documentation about the garbage collector.

Reply:
I asked the compiler team whether or not the `while time() < t0 end` loop could ever be removed, and it couldn't. It should thus be a safe implementation

Reply:
> Do you think it's a good idea ?

I'd favor a warning in the docs and let the handling of the GC be done by the user. If the user has a loop that is sensitive enough for GC to matter, they probably need a better sleeping utility than `sleep` and more fine grained control control over GC anyways.